ibis/expr/analysis.py

```diff
@@ -201,7 +201,10 @@ def substitute(fn, expr):
     """
     Substitute expressions with other expressions.
     """
-    node = expr.op()
+    try:
```

Comment:
Is this required?

Reply:
Only if we merge this before #4015. I'll remove it so this PR fails until we merge that one.